### DISCUSSION TITLE
Questions regarding linked data sets

### DISCUSSION THREAD

**Discussion by saskra** [JacobBumgarner/VesselVio#21] [Q&A] [answered]
In your segmentation tips https://github.com/JacobBumgarner/VesselVio#segmentation-tips you link to Vessap, there are among other things also pre-segmented brains for download. Unfortunately, these cause Vesselvio to crash when I try to analyze them. First other programs on the computer no longer work and then the VesselVio window simply disappears. Have you ever tested this as well, and do you think it's more likely due to too weak hardware, or does your software have a general limitation up to a certain file size?
Also, I'd be interested to know if there wouldn't be a way to load the already skeletonized brains offered there as well directly into VesselVio.

Furthermore, you also link to ClearMap/TubeMap in this section, there is even a ready graph available for download: https://osf.io/dtwfg/?view_only=4427a838cbd0468c9fbad9cab465d866 Although VesselVio offers the import of graphs, this one seems to be in the wrong file format. Is there a way to get it in there anyway?

**Comment by JacobBumgarner** [comment, marked as answer]
For the first question, what is exact file you’re trying to run through VesselVio?

I analyzed the whole brain BalbC datasets on my MacBook Pro laptop, so I don’t imagine that it’s an issue with the size. What’re the specs are your computer? It’s possible that if you’re trying to analyze the whole brain at once that your computer is running out of memory.

---

And for the graph: I’ll give it a download when I’m on my computer today to see what the file type is and if it would be worth adding compatibility with that file type.

Thanks for being so active with exploring the program! 😁

**Reply by saskra** [reply]
I'm testing the smallest file CD1-E_no2-iso3um_stitched_segmentation.nii, unpacked it's 25 GB - so it's even very likely that my 16 GB RAM on the Windows machine is not enough for this. I will then test it on my Linux server, there of course I am missing the graphics. Did you analyze the whole brain at once or in small pieces bit by bit?

**Reply by JacobBumgarner** [reply]
I analyzed individual regions rather than the whole brain all at once. You can do this type of analysis on your 16GB RAM machine. Just load the `CD1-E_no2-iso3um_stitched_annotation_result.nii` file (can’t remember exact name) alongside an annotation file created on the `Annotation Processing` page. The app will then created a cached annotation file and will memory map it for region access so that you don’t have to have 25 GB of ram

On the other hand, if you want to analyze the whole brain, I would run the v1.1.2-dev release from the terminal (make sure to install the new `requirements.txt`, `igraph` got a huge speed bump in their latest update). Just recognize that the skeletonization of the whole brain is going to take a long time. I’ve never actually run the test to see _how_ long… but it’ll take a while.

I make this point in the limitations discussion of the paper. In my opinion, analyzing the whole brain at once doesn’t answer any specific research questions, so I didn’t end up running or testing this type of ‘whole brain at once’ analysis. Hopefully that makes sense

**Reply by saskra** [reply]
> On the other hand, if you want to analyze the whole brain, I would run the v1.1.2-dev release from the terminal (make sure to install the new `requirements.txt`, `igraph` got a huge speed bump in their latest update). Just recognize that the skeletonization of the whole brain is going to take a long time. I’ve never actually run the test to see _how_ long… but it’ll take a while.

Thanks, took about 5 hours, so an acceptable amount of time.

> I make this point in the limitations discussion of the paper. In my opinion, analyzing the whole brain at once doesn’t answer any specific research questions, so I didn’t end up running or testing this type of ‘whole brain at once’ analysis. Hopefully that makes sense

I actually agree with that. It's just a "everyone wants this right now, so do I" thing. ;-)

**Comment by saskra** [comment]
I have another question about this section of your tutorial: which of the three linked packages did you end up using the most for segmentation? They seem to be quite different and not always easy to get working, and the segmentation there is always only a small part of a larger pipeline and not so easy to cut out. Is there an automatic link of a segmentation algorithm to VesselVio? And have you by chance encountered a solution if the blood vessel images were created with a filling stain instead of a hollow vessel wall stain?

**Reply by JacobBumgarner** [reply]
Hi @saskra,
Unfortunately I won’t be able to provide much help/guidance for those linked pipelines. All of my datasets are filled-vessels, so my I’ve never had a need for CNN-based segmentation. I can always segment my datasets using pure image processing techniques. I’ve linked those pipelines mainly because I’ve heard about/seen second hand success with using them.

Since you’ve acquired filled-vessel datasets, I’d be happy to spend 30 mins or so walking through some of my segmentation approaches with you. Feel free to email me.

**Reply by saskra** [reply]
Thanks for the quick reply! Then I'll stick with my current classical-algorithmic segmentation solution as well. That was another case of "if everyone is doing Deep Learning, it just has to be better somehow." ^^

**Reply by JacobBumgarner** [reply]
You bet!

Yes, deep learning has high FOMO appeal truthfully, but fine-tuned algorithmic approaches are very hard to beat more often than not. 😁

Still happy to chat about approaches either way.

If I had infinite time, I’d love to add an image processing component to VesselVio where users could segment their datasets and then automatically move them over to be analyzed/visualized. Ideally the segmentation page would expose many of the scipy & scitkit-image algorithms, provide access to some CNN models, and provide pre-baked segmentation pipelines.

Maybe one day after I’m done refactoring and automating the codebase lol.
